src/HtmlAgilityPack.Shared/HtmlWeb.cs

```diff
@@ -801,6 +802,15 @@ internal static HttpClient GetSharedHttpClient(string userAgent)
         /// <value>The automatic decompression.</value>
         public DecompressionMethods AutomaticDecompression { get; set; }
 
+        /// <summary>
+        /// Gets or sets the timeout value in milliseconds.
+        /// </summary>
+        public int Timeout
+        {
+            get { return _timeout; }
+            set { if (value < 0) { _timeout = 0; } else { _timeout = value; } }
```

Comment:
Why should an assignment like `HtmlWeb.Timeout = -4567` be equivalent to `HtmlWeb.Timeout = 0`?

In my opinion it would rather make more sense to throw an _ArgumentOutOfRangeException_ for negative TimeOut values, the same behavior already established by [_HttpWebRequest.Timeout_](https://learn.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest.timeout) and [_HttpClient.Timeout_](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout).

Reply:
Good idea, I adjusted the setter to throw a ArgumentOutOfRangeException if the value is less than or equal to zero. Thanks!

Reply:
Note that i edited my comment, as i originally did not account for [Timeout.Infinite](https://learn.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinite?system-threading-timeout-infinite) (which is the value -1)

So, basically, the setter should throw if `value < 0 && value != Timeout.Infinite`.

(My apologies for this oversight in my original comment.)

Reply:
To be honest, I do not expect any users to be waiting infinitely for a document to load. The default value for `HttpClient.Timeout` and `HttpWebRequest.Timeout` is 100,000 ms, or 100 seconds. Adding an infinite timeout is a bit troublesome since `HttpWebRequest` requires an integer while `HttpClient` requires a `Timespan` object to be passed instead. I think throwing an exception if the value is zero or less is more of a practical approach.

Reply:
> Adding an infinite timeout is a bit troublesome since `HttpWebRequest` requires an integer while `HttpClient` requires a `Timespan` object to be passed instead.

No, it's actually not troublesome :-) because [_Timeout.InfiniteTimeSpan_](https://learn.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinitetimespan) is per definition a TimeSpan value of -1 millisecond. Therefore`TimeSpan.FromMilliseconds(-1)` equals `Timeout.InfiniteTimeSpan`.

Reply:
Oh really? I thought `TimeSpan.FromMilliseconds(-1)` would throw an exception that it's out of bounds. In this case, we can change that if statement to be this `if (value <= 0 && value != -1) { throw new ArgumentOutOfRangeException(); }` There you go, works as expected.